sdk/openai/azure-openai/tests/conftest.py

```diff
@@ -77,38 +70,26 @@ def client(api_type, api_version):
     if api_type == "azure":
         client = openai.AzureOpenAI(
             azure_endpoint=os.getenv(ENV_AZURE_OPENAI_ENDPOINT),
-            api_key=os.getenv(ENV_AZURE_OPENAI_KEY),
+            azure_ad_token_provider=get_bearer_token_provider(DefaultAzureCredential(), "https://cognitiveservices.azure.com/.default"),
```

Comment:
Does this work?

It should fail in pipeline and recordings?

Reply:
We only do live testing. Yes, it works and verifying changes again in the pipeline linked in the PR description

Reply:
Do you need DefaultAzureCredential(exclude_managed_identity_credential=True)?

Reply:
https://gist.github.com/xiangyan99/9afccaf809f36382ed306c825e9bad3b#why-we-disable-managed-identity-in-dac

Reply:
I didn't need it yesterday when testing, but I'll check the results of the currently running pipeline to see if it needs to be added.

Reply:
Few flaky tests failing, but otherwise good. Appears I don't need it?

Reply:
I saw

2024-06-05T18:41:14.3646600Z WARNING  azure.identity._credentials.chained:chained.py:123 DefaultAzureCredential failed to retrieve a token from the included credentials.
2024-06-05T18:41:14.3648640Z Attempted credentials:
2024-06-05T18:41:14.3650510Z 	EnvironmentCredential: EnvironmentCredential authentication unavailable. Environment variables are not fully configured.
2024-06-05T18:41:14.3652660Z Visit https://aka.ms/azsdk/python/identity/environmentcredential/troubleshoot to troubleshoot this issue.
2024-06-05T18:41:14.3654810Z 	ManagedIdentityCredential: ManagedIdentityCredential authentication unavailable, no response from the IMDS endpoint.
2024-06-05T18:41:14.3657050Z 	SharedTokenCacheCredential: SharedTokenCacheCredential authentication unavailable. No accounts were found in the cache.
2024-06-05T18:41:14.3659200Z 	AzureCliCredential: Please run 'az login' to set up an account
2024-06-05T18:41:14.3661010Z 	AzurePowerShellCredential: Failed to invoke PowerShell.
2024-06-05T18:41:14.3663130Z To mitigate this issue, please refer to the troubleshooting guidelines here at https://aka.ms/azsdk/python/identity/powershellcredential/troubleshoot.
2024-06-05T18:41:14.3665940Z 	AzureDeveloperCliCredential: Azure Developer CLI could not be found. Please visit https://aka.ms/azure-dev for installation instructions and then,once installed, authenticate to your Azure account using 'azd auth login'.

in one of your failed runs?